rs/cli/src/auth.rs

```diff
@@ -156,10 +106,60 @@ impl Auth {
         }
         Ok(None)
     }
+}
 
-    pub async fn auto_detect_neuron_id(&self, nns_urls: &[url::Url]) -> anyhow::Result<u64> {
-        let sender = match self {
-            Auth::Hsm { pin, slot, key_id } => {
+#[derive(Clone, Debug)]
+pub struct Auth {
+    pub auth_type: AuthType,
+    maybe_neuron: Option<Neuron>,
+}
+
+impl Auth {
+    pub fn new(auth_type: AuthType) -> Self {
+        Self {
+            auth_type,
+            maybe_neuron: None,
+        }
+    }
+
+    pub fn with_neuron(self, neuron: Neuron) -> Self {
+        Self {
+            maybe_neuron: Some(neuron),
+            ..self
+        }
+    }
```

Comment:
Kind of against allowing this since this means we can always override it after initialization. The idea for safe execution would be to initialize within `ctx.rs` and after that you can only use what you configured. This means that magically someone from somewhere may mutate this specific thing is what I tried to avoid. After the call to construct the `DreContext` one should only use what he configured. No mutations or translations. This feels like a hack